File: swingLotto/src/main/java/com/eastgrams/swinglotto/service/LottoDrawResultService.java
package com.eastgrams.swinglotto.service;

import com.eastgrams.swinglotto.model.LottoDrawServiceResult;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface LottoDrawResultService {
    @GET("common.do?method=getLottoNumber")
    Call<LottoDrawServiceResult> getLottoResult(
            @Query("drwNo") int drawNo
    );
}

File: android_lottoTV/app/src/main/java/com/eastgrams/lottoResultTvApp/service/LottoDrawResultService.kt
package com.eastgrams.lottoResultTvApp.service

import com.eastgrams.lottoResultTvApp.data.LottoDrawServiceResult
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Query
import kotlin.jvm.java

interface LottoDrawResultService {
    @GET("common.do")
    suspend fun getLottoResult(
        @Query("method") method: String = "getLottoNumber",
        @Query("drwNo") drawNo: Int,
    ): LottoDrawServiceResult

    companion object {
        private const val BASE_URL = "https://www.dhlottery.co.kr/"
        fun create(): LottoDrawResultService {
            val logger = HttpLoggingInterceptor().apply { level = HttpLoggingInterceptor.Level.BASIC }

            val client =  OkHttpClient.Builder()
                .addInterceptor(logger)
                .build()

            return Retrofit.Builder()
                .baseUrl(BASE_URL)
                .client(client)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
                .create(LottoDrawResultService::class.java)
        }
    }
}